Python/ibmcloudsql/catalog_table.py

```diff
@@ -0,0 +1,311 @@
+        ----------
+        table_name: str
+            The name of the catalog table
+
+        cos_url : str, optional
+            The COS URL from which the catalog table should reference to
+            If not provided, it uses the internal `self.cos_in_url`
+
+        force_recreate: bool, optional
+            (True) force to recreate an existing catalog table
+        blocking: bool, optional
+            (True) wait until it returns the resut
+
+        Returns
+        -------
+            none if job "failed"
+            otherwise returns
+        """
+        def get_catalog_table_async(
+                                    table_name,
+                                    cos_url=None,
+                                    force_recreate=False):
+            """
+            the async version of `get_catalog_table`
+
+            Parameters
+            ----------
+            table_name: str
+                The name of the catalog table
+
+            cos_url : str, optional
+                The COS URL from which the catalog table should reference to
+                If not provided, it uses the internal `self.cos_in_url`
+
+            force_recreate: bool, optional
+                (True) force to recreate an existing catalog table
+
+            Returns
+            ------
+                job_id [if the table is being created]
+                None   [if the table already created]
+            """
+            self.current_catalog_table_name = table_name
+
+            if cos_url is None:
+                cos_url = self.cos_in_url
+
+            # from IPython.display import display
+            df = self.show_catalog_tables()
+            try:
+                found = df[df['tableName'].str.contains(table_name)]
+            except Exception:
+                # not found
+                found = []
+
+            # if logger.getEffectiveLevel() == logging.DEBUG:
+            #     display(df)
+            if len(found) > 0 and force_recreate:
+                self.drop_catalog_table(table_name)
+            self.regular_tables.add(table_name)
+            if len(found) == 0 or force_recreate:
+                sql_stmt_create = self.sql_stmt_create_template.format(
+                    table_name=table_name, cos_in=cos_url)
+                logger.debug(sql_stmt_create)
+                job_id = self.submit_sql(sql_stmt_create)
+                return job_id
+            return None
+
+        job_id = get_catalog_table_async(table_name,
+                                              cos_url,
+                                              force_recreate=force_recreate)
+        if job_id is not None and blocking is True:
+            job_status = self.wait_for_job(job_id)
+            if job_status == "completed":
+                return self.get_result(job_id)
+            else:
+                return None
+        return None
+
+    def get_catalog_table_partitioned(self, table_name, cos_url=None, force_recreate=False):
```

Comment:
create_partitioned_table

Reply:
Updated in commit `00fc7b5bd48668386d01dad60f89b1069c8f8f96`